code/modules/xenomorph/xeno_structures.dm

```diff
@@ -127,36 +119,21 @@
 	COOLDOWN_START(src, silo_damage_alert_cooldown, XENO_HEALTH_ALERT_COOLDOWN) //set the cooldown.
 
 ///Alerts the Hive when hostiles get too close to their resin silo
-/obj/structure/resin/silo/proc/resin_silo_proxy_alert(datum/source, atom/hostile)
-	SIGNAL_HANDLER
-
+/obj/structure/resin/silo/proc/resin_silo_proxy_alert()
 	if(!COOLDOWN_CHECK(src, silo_proxy_alert_cooldown)) //Proxy alert triggered too recently; abort
 		return
-
-	if(!isliving(hostile))
-		return
-
-	var/mob/living/living_triggerer = hostile
-	if(living_triggerer.stat == DEAD) //We don't care about the dead
-		return
-
-	if(isxeno(hostile))
-		var/mob/living/carbon/xenomorph/X = hostile
-		if(X.hive == associated_hive) //Trigger proxy alert only for hostile xenos
-			return
-
-	if(get_dist(loc, hostile) > 2) //Can only send alerts for those within 2 of us; so we don't have all silos sending alerts when one is proxy tripped
+	var/mob/living/carbon/human/nearby_human
+	for (var/mob/living/carbon/human/human as anything in cheap_get_humans_near(src, XENO_SILO_DETECTION_RANGE))
+		if(human.stat != DEAD)
+			nearby_human = human
+			break
+		
+	if(!nearby_human)
 		return
```

Comment:
I don't think this check is necessary. From what I've read [here](https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#avoid-unnecessary-type-checks-and-obscuring-nulls-in-lists) it seems like there won't be any nulls in that loop.

Reply:
it's null if there's nothing in cheap_get_humans_near, which happens 99% of the time